File: android/src/main/java/expo/modules/adaptiveglass/ExpoAdaptiveGlassGroupModule.kt
package expo.modules.adaptiveglass

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

// own module, older Expo SDKs allow one view per module
class ExpoAdaptiveGlassGroupModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoAdaptiveGlassGroup")

    View(GlassGroupView::class) {
      Prop("spacing") { view: GlassGroupView, value: Double -> view.spacingDp = value.toFloat() }
      Prop("intensity") { view: GlassGroupView, value: Double -> view.intensity = value.toFloat() }
      Prop("tintColor") { view: GlassGroupView, value: Int? -> view.tint = value }
      Prop("tintScheme") { view: GlassGroupView, value: String -> view.tintScheme = value }

      OnViewDidUpdateProps { view: GlassGroupView ->
        view.propsDidUpdate()
      }
    }
  }
}

File: ios/ExpoAdaptiveGlassGroupModule.swift
import ExpoModulesCore

// separate module: older SDKs allow one view per module
public class ExpoAdaptiveGlassGroupModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoAdaptiveGlassGroup")

    View(GlassGroupView.self) {
      Prop("spacing") { (view, value: Double) in view.spacing = CGFloat(value) }
      Prop("intensity") { (view, value: Double) in view.intensity = CGFloat(value) }
      Prop("tintColor") { (view, value: UIColor?) in view.glassTint = value }
      Prop("tintScheme") { (view, value: String) in view.tintScheme = value }

      OnViewDidUpdateProps { view in
        view.propsDidUpdate()
      }
    }
  }
}
